lib/hypervisor/hv_kvm/monitor.py

```diff
@@ -538,9 +538,15 @@ def HotAddDisk(self, disk, devid, uri, drive_add_fn=None):
 
     """
     if os.path.exists(uri):
-      fd = os.open(uri, os.O_RDWR)
-      fdset = self._AddFd(fd)
-      os.close(fd)
+      # For QEMU's dynamic auto-read-only feature to work (on by default since
+      # QEMU 4.0), we need two file descriptors, one in read-write and one in
+      # read-only mode.
```

Comment:
I would add here a note that fdsets work with older versions; the wording right now is unclear on this.

Reply:
Agreed